### PR TITLE
Use std::map as power policy registry

### PR DESCRIPTION
**Pull request by wentasah**
This allows to print the list of available policies to the user in an
error message.

### CODE REVIEW

File: src/power_policy/_power_policy.cpp [hunk 1]
```diff
@@ -50,16 +65,15 @@ static std::unique_ptr<PowerPolicy> instantiate_power_policy(
...
-    PP("imx8_fixed", PowerPolicy_Imx8_Fixed, 2);
-
-    throw std::runtime_error("Unknown power policy selected: " + policy_name);
+    try {
+        return power_policies.at(policy_name)(policy_args);
+    } catch (const std::out_of_range &) {
+        using namespace std::string_literals;
+        std::string all_pp;
+        for (auto const& [pp, val] : power_policies)
+            all_pp += (all_pp.empty() ? "" : ", "s) + pp;
```
Comment (MatejKafka) [inline src/power_policy/_power_policy.cpp:74]: why the `""s` suffix? I think it's not needed here.

File: src/power_policy/_power_policy.cpp [hunk 2]
```diff
@@ -20,17 +22,30 @@
...
+            if (arg_count != policy_args.size()) { /* NOLINT(readability-container-size-empty) */       \
+                throw runtime_error(                                                                    \
+                    "Incorrect number of parameters for the '" name "' power policy; got " +            \
+                    std::to_string(policy_args.size()) + ", expected " #arg_count);                     \
+            }                                                                                           \
+            return std::make_unique<class_>(ARR_ARGS_##arg_count(policy_args));                         \
+        }                                                                                               \
+    }
+
+static const std::map<std::string, std::function<std::unique_ptr<PowerPolicy>(std::vector<std::string>)>> power_policies {
```
Comment (MatejKafka) [inline src/power_policy/_power_policy.cpp:37]: Missing `const ... &` (I think the vector gets copied, and then reference to the copy passed to the lambda)

File: src/power_policy/_power_policy.cpp (hunk 1 above)
Comment (wentasah) [reply src/power_policy/_power_policy.cpp:74]: Leftover from a previous version :-)

File: src/power_policy/_power_policy.cpp (hunk 2 above)
Comment (wentasah) [reply src/power_policy/_power_policy.cpp:37]: Probably yes, thanks!

**Pull request merged**